engine/src/main/java/com/datatorrent/stram/util/LoggerUtil.java

```diff
@@ -464,4 +470,68 @@ public static boolean removeAppender(String name)
     }
     return names;
   }
+
+  /**
+   * Makes MDC properties
+   */
+  private static void makeMDCProperties()
+  {
+    if (setMDCProperties) {
+
+      MDC.put("node", getCurrentHost());
+
+      String value = System.getenv("HADOOP_USER_NAME");
+      if (value != null) {
+        MDC.put("user", value);
+      }
+
+      value = System.getenv("CONTAINER_ID");
+      if (value != null) {
+        MDC.put("containerId", value);
+      }
+
+      Properties properties = System.getProperties();
+      String longName = LogicalPlan.APPLICATION_NAME.getLongName();
+      if (properties.contains(longName)) {
+        MDC.put("application", properties.getProperty(longName));
+      }
+
+      if (MDC.get("service") == null) {
+        MDC.put("service", getService());
+      }
+
+      setMDCProperties = false;
+    }
+  }
+
+  /**
+   * Returns the service name
+   * @return Service name
+   */
+  private static String getService()
+  {
+    StackTraceElement[] stack = Thread.currentThread().getStackTrace();
+    String mainClass = stack[stack.length - 1].getClassName();
+    if (mainClass.equals("com.datatorrent.stram.cli.ApexCli")) {
+      return "apexclient";
+    } else if (mainClass.equals("com.datatorrent.stram.StreamingAppMaster")) {
+      return "apexmaster";
+    } else if (mainClass.equals("com.datatorrent.stram.engine.StreamingContainer")) {
+      return "apexcontainer";
```

Comment:
I agree with renaming of the service names and the usage the real class names instead of string constants.

But I'd prefer to have the current implicit logic for definition of the service name. If somebody needs to set the service name explicitly I can provide a method to do it.